netbox/dcim/models/devices.py

```diff
@@ -1062,3 +1063,83 @@ def delete(self, *args, **kwargs):
             )
 
         return super().delete(*args, **kwargs)
+
+
+class VirtualDeviceContext(PrimaryModel):
+    device = models.ForeignKey(
+        to='Device',
+        on_delete=models.PROTECT,
+        related_name='vdcs',
+        blank=True,
+        null=True
+    )
+    name = models.CharField(
+        max_length=64
+    )
+    status = models.CharField(
+        max_length=50,
+        blank=True,
```

Comment:
IMO `status` should be a required field, as with other models.

Reply:
Going to make this change now